compiler/symbol.go

```diff
@@ -375,19 +375,26 @@ func (c *compilerContext) checkWasmImport(f *ssa.Function, pragma string) {
 
 // Check whether the type maps directly to a WebAssembly type, according to:
 // https://github.com/golang/go/issues/59149
+// TODO(ydnar): document why we relaxed this for WASI Preview 2.
 func isValidWasmType(typ types.Type, isReturn bool) bool {
 	switch typ := typ.Underlying().(type) {
 	case *types.Basic:
 		switch typ.Kind() {
-		case types.Int32, types.Uint32, types.Int64, types.Uint64:
+		case types.Bool:
+			return true
+		case types.Int8, types.Uint8, types.Int16, types.Uint16, types.Int32, types.Uint32, types.Int64, types.Uint64:
 			return true
 		case types.Float32, types.Float64:
 			return true
-		case types.UnsafePointer:
-			if !isReturn {
-				return true
-			}
+		case types.Uintptr, types.UnsafePointer:
```

Comment:
These will be 32 bits wide in GOARCH=wasm32.

Reply:
I see. It'll still be slightly incompatible but only because TinyGo uses 32-bit pointers in GOOS=wasm (so GOOS=wasm in TinyGo behaves like GOOS=wasm32). Seems fine by me.